File: solutions/python/src/adapters/persistence/in_memory_authorization_service.py
"""Local whitelist: the site must keep authorizing when HQ is unreachable (§1.1)."""


class InMemoryAuthorizationService:
    def __init__(self, valid_tags: list[str] | None = None) -> None:
        self._known = set(valid_tags or ["TAG-MONTHLY-77"])

    def is_authorized(self, id_tag: str) -> bool:
        return id_tag in self._known

File: solutions/node/src/adapters/persistence/InMemoryAuthorizationService.ts
import type { AuthorizationService } from '../../charging/application/ports.ts';

// Local whitelist: the site must keep authorizing when HQ is unreachable (§1.1).
export class InMemoryAuthorizationService implements AuthorizationService {
  private readonly known: Set<string>;
  constructor(validTags: string[] = ['TAG-MONTHLY-77']) {
    this.known = new Set(validTags);
  }
  async isAuthorized(idTag: string): Promise<boolean> {
    return this.known.has(idTag);
  }
}
